flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -143,7 +147,26 @@ public void run(SourceContext<FlinkInputSplit> ctx) throws Exception {
     }
   }
 
-  private void monitorAndForwardSplits() {
+  private long maxReachableSnapshotId(long lastConsumedSnapshotId, long latestSnapshotId,
+                                      int maxSnapshotCountPerMonitorInterval) {
+    // This doesn't consider snapshot inheritance since it is already checked in state initialization.
```

Comment:
What do you mean by "snapshot inheritance" here?

Reply:
The latest table snapshot id might not be the ancestor of the last consumed one.